src/os/bluestore/BlueStore.cc

```diff
@@ -12399,6 +12399,40 @@ void BlueStore::_zoned_clean_zone(uint64_t zone_num) {
   dout(10) << __func__ << " cleaning zone " << zone_num << dendl;
   // TODO: (1) copy live objects from zone_num to a new zone, (2) issue a RESET
   // ZONE operation to the device for the corresponding zone.
+
+  zone_state_t zone_state;
+  std::string pfx = _zoned_get_prefix(zone_num * bdev->get_zone_size());
+  KeyValueDB::Iterator it = db->get_iterator(pfx, KeyValueDB::ITERATOR_NOCACHE);
```

Comment:
IMO you should revise naming scheme a bit for the same of uniformity with other stuff.
Prefix to be a single char field, e.g. 'G'
And the key itself to includes zone + oid + whatever...
Hence one will need to call it->lower_bound(key(zone)) after get_iterator to locate the first record belonging to a desired zone.

Reply:
This one is resolved in https://github.com/ceph/ceph/pull/41845